velox/exec/Task.cpp

```diff
@@ -418,6 +420,51 @@ void Task::removeSpillDirectoryIfExists() {
   }
 }
 
+/// Builds the query trace config.
+std::optional<QueryTraceConfig> Task::maybeMakeTraceConfig() const {
+  const auto& queryConfig = queryCtx_->queryConfig();
+  if (!queryConfig.queryTraceEnabled()) {
+    return std::nullopt;
+  }
+
+  const auto queryTraceDir = queryConfig.queryTraceDir();
```

Comment:
I think we shall attach task id to the query config as we might run multiple tasks in parallel in different nodes and each one has its own directory. Also we shall add a step on task creation to delete the directory first to cleanup the potential leftover from previous run.